v2/tools/generator/internal/codegen/pipeline/export_controller_type_registrations.go

```diff
@@ -30,32 +30,38 @@ func ExportControllerResourceRegistrations(idFactory astmodel.IdentifierFactory,
 
 			var resources []astmodel.TypeName
 			var storageVersionResources []astmodel.TypeName
+			var resourceExtensions []astmodel.TypeName
 			indexFunctions := make(map[astmodel.TypeName][]*functions.IndexRegistrationFunction)
 			secretPropertyKeys := make(map[astmodel.TypeName][]string)
 
 			// We need to register each version
 			for _, def := range types {
-				resource, ok := astmodel.AsResourceType(def.Type())
-				if !ok {
-					continue
-				}
 
-				if resource.IsStorageVersion() {
-					storageVersionResources = append(storageVersionResources, def.Name())
+				if resource, ok := astmodel.AsResourceType(def.Type()); ok {
+
+					if resource.IsStorageVersion() {
+						storageVersionResources = append(storageVersionResources, def.Name())
 
-					chains, err := catalogSecretPropertyChains(def, types)
-					if err != nil {
-						return nil, errors.Wrapf(err, "failed to catalog %s property chains", def.Name())
+						chains, err := catalogSecretPropertyChains(def, types)
+						if err != nil {
+							return nil, errors.Wrapf(err, "failed to catalog %s property chains", def.Name())
+						}
+
+						resourceIndexFunctions, resourceSecretPropertyKeys := handleSecretPropertyChains(chains, idFactory, def)
+						indexFunctions[def.Name()] = resourceIndexFunctions
+						secretPropertyKeys[def.Name()] = resourceSecretPropertyKeys
 					}
 
-					resourceIndexFunctions, resourceSecretPropertyKeys := handleSecretPropertyChains(chains, idFactory, def)
-					indexFunctions[def.Name()] = resourceIndexFunctions
-					secretPropertyKeys[def.Name()] = resourceSecretPropertyKeys
+					resources = append(resources, def.Name())
+				} else if resourceExtension, ok := astmodel.AsObjectType(def.Type()); ok {
```

Comment:
minor: you don't know it's a `resourceExtension` at this point, it's just an `objectType`?

Reply:
Yep, you're right. Made it to be called `object`.